src/formatDate.js

```diff
@@ -50,7 +50,27 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const arrFromDate = date.split(oldSeparator);
+  const arrResultDate = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
```

Comment:
Nested for-loops have a high time complexity (O(n^2)). Consider using a more efficient algorithm or data structure, like a HashMap, to improve performance.